### PR TITLE
chore: optimise code around the screen curtain logic [WPB-7306]

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7306" title="WPB-7306" target="_blank">StoryWPB-7306</a>  [iOS] : Add “Conversation” , “Archive” , “Settings”
  </td></table>
  <br />
 

### Issue

We have two view controllers named `RootViewController`, one nested in `ScreenCurtain`. This is confusing when debugging the view hierarchy.
This PR converts a view controller which loads its view from a nib into a SwiftUI view.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios-utilities/Source/Public/DispatchGroupQueue.swift [hunk 1]
```diff
@@ -26,15 +26,15 @@ public final class DispatchGroupQueue: NSObject, ZMSGroupQueue {
...
     public init(queue: DispatchQueue) {
         self.queue = queue
-        self.dispatchGroupContext = DispatchGroupContext(groups: [])
+        dispatchGroupContext = DispatchGroupContext(groups: [])
     }
 
-    public  var dispatchGroup: ZMSDispatchGroup! {
-        return self.dispatchGroupContext.groups.first
+    public  var dispatchGroup: ZMSDispatchGroup {
+        dispatchGroupContext.groups[0]
```
Comment (netbe) [inline wire-ios-utilities/Source/Public/DispatchGroupQueue.swift]: why the change ? shouldn't we keep first and have it optional?

File: wire-ios/Wire-iOS/Sources/UserInterface/LaunchImage/LaunchImageViewController.swift [hunk 2]
```diff
@@ -43,26 +43,21 @@ class LaunchImageViewController: UIViewController {
...
-        nibView?.translatesAutoresizingMaskIntoConstraints = false
-        if let nibView = nibView {
-            view.addSubview(nibView)
-        }
-        if let nibView = nibView {
-            contentView = nibView
-        }
+        let contentView = UINib(nibName: "AppLock", bundle: nil)
+            .instantiate(withOwner: nil, options: nil)
+            .first as! UIView
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/LaunchImage/LaunchImageViewController.swift]: I would have kept the helper method on UIView since it's used twice

File: wire-ios/Wire-iOS/Sources/UserInterface/ScreenCurtain/ScreenCurtainView.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import SwiftUI
+
+struct ScreenCurtainView: View {
+
+    var body: some View {
+        VStack {
+            Spacer()
+            HStack {
+                Spacer()
+                // for some reason `Image(.Wire.shield)` didn't show the shield
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/ScreenCurtain/ScreenCurtainView.swift:28]: on preview or device?

File: wire-ios-utilities/Source/Public/DispatchGroupQueue.swift (hunk 1 above)
Comment (caldrian) [reply wire-ios-utilities/Source/Public/DispatchGroupQueue.swift]: The implicit force-unwrap results from an unspecified nullability in Objective C code. Since we try not to use force-unwraps when possible we should either declare `dispatchGroup` as optional or not. 
I will have another look.

File: wire-ios/Wire-iOS/Sources/UserInterface/ScreenCurtain/ScreenCurtainView.swift (hunk 3 above)
Comment (caldrian) [reply wire-ios/Wire-iOS/Sources/UserInterface/ScreenCurtain/ScreenCurtainView.swift:28]: On the device. It's weird, I spent a while finding the cause why the image wasn't visible.

File: wire-ios-utilities/Source/Public/DispatchGroupQueue.swift (hunk 1 above)
Comment (caldrian) [reply wire-ios-utilities/Source/Public/DispatchGroupQueue.swift]: Fixed optionality in the protocol and where needed: 068fb1c9f59272bde53e8a0f84cc86506c6d1752

File: wire-ios/Wire-iOS/Sources/UserInterface/ScreenCurtain/ScreenCurtainView.swift (hunk 3 above)
Comment (caldrian) [reply wire-ios/Wire-iOS/Sources/UserInterface/ScreenCurtain/ScreenCurtainView.swift:28]:
```
Image(.Wire.shield)
```

Screenshot 2024-05-06 at 9 48 59 AM

```
Image(uiImage: .init(resource: .Wire.shield))
```

Screenshot 2024-05-06 at 9 49 29 AM

File: wire-ios/Wire-iOS/Sources/UserInterface/LaunchImage/LaunchImageViewController.swift (hunk 2 above)
Comment (caldrian) [reply wire-ios/Wire-iOS/Sources/UserInterface/LaunchImage/LaunchImageViewController.swift]: Since a static method `UIView.shieldView()` is not what we want, I created a temporary helper: 4471295fd78e12893e8726d2557f040d45b6217a
On the long run I want to look into the `LaunchImageViewController` and if it's really needed.

**Review by netbe — APPROVED** [review]
LGTM;)

File: wire-ios/Wire-iOS/Sources/UserInterface/ScreenCurtain/ScreenCurtainView.swift (hunk 3 above)
Comment (netbe) [reply wire-ios/Wire-iOS/Sources/UserInterface/ScreenCurtain/ScreenCurtainView.swift:28]: ok

**Review by agisilaos — APPROVED** [review]
